### PR TITLE
CAM2-206 Campus.il Makes the instance types used by asgard configurable

### PR DESCRIPTION
**Pull request by pomegranited**
Adds `HOTG_ADDITIONAL_INSTANCE_TYPES`, which can be used to expand the list of instance types allowed when creating clusters on Asgard.

Also adds the `supervisor` role as a meta requirement, as this is required to run the `hotg` role.

**Merge deadline**: None

**Testing instructions**:

This is the deployment process we used for the updated Campus admin server, so this change can be verified by checking the [olive admin instance type list](http://admin.edx-flatu.org/eu-west-1/instanceType/list).

1. Add an ansible variable like:
   ```
   HOTG_ADDITIONAL_INSTANCE_TYPES:
     t3.xlarge:
       price: 0.052
       family: 'Burstable'
       group: 't3'
       size: 'Extra Large'
       vCpu: '4'
       ecu: '6'
       mem: '16.00'
   ```
1. Deploy asgard using the [vpc_admin.yml](https://github.com/edx/configuration/blob/master/playbooks/vpc_admin.yml) playbook.
1. Check the Asgard EC2 > Instance Types list to ensure the additional types are added.

**Author notes and concerns**:

1. Simply installing Asgard on a fresh instance added some new instance types (e.g. `c5.xlarge`), but the `t3.xlarge` type was not added.  So this change ensures that it is always added, since we rely on it for olive's `edxapp` cluster.

**Reviewers**
- [ ] @swalladge
- [ ] edX reviewer[s] TBD

Configuration Pull Request
---

Make sure that the following steps are done before merging:

  - [ ] A DevOps team member has approved the PR if it is code shared across multiple services and you don't own all of the services.
  - [ ] Are you adding any new default values that need to be overridden when this change goes live? If so:
    - [ ] Update the appropriate internal repo (be sure to update for all our environments)
    - [ ] If you are updating a secure value rather than an internal one, file a DEVOPS ticket with details.
    - [ ] Add an entry to the CHANGELOG.
  - [ ] If you are making a complicated change, have you performed the proper testing specified on the [Ops Ansible Testing Checklist](https://openedx.atlassian.net/wiki/display/EdxOps/Ops+Ansible+Testing+Checklist)?  Adding a new variable does not require the full list (although testing on a sandbox is a great idea to ensure it links with your downstream code changes).
  - [ ] Think about how this change will affect Open edX operators.  Have you updated the wiki page for the next Open edX release?

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @pomegranited! I've created [OSPR-3788](https://openedx.atlassian.net/browse/OSPR-3788) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

**Comment by natabene** [comment]
@pomegranited Thank you for your contribution. Please let me know once it is ready for our review.

**Review by samuelallan72 — APPROVED** [review]
:+1: with query: should the format of the `HOTG_ADDITIONAL_INSTANCE_TYPES` option be documented there, since it's a more complex structure?

- [X] I tested this:
  - verified that the asgard instance deployed with this includes the extra instance types
- [X] I read through the code
- [ ] ~I checked for accessibility issues~
- [ ] Includes documentation

**Comment by pomegranited** [comment]
@swalladge 

> should the format of the HOTG_ADDITIONAL_INSTANCE_TYPES option be documented there, since it's a more complex structure?

Good point, done with b85529a 

@natabene this is ready for review.  It is not urgent, so happy to wait until next month's review budget.  Thank you!

File: playbooks/roles/hotg/defaults/main.yml [hunk 1]
```diff
@@ -51,6 +51,18 @@ HOTG_CALLBACK_URI: "{{ HOTG_URL }}/auth/signIn"
...
+# HOTG_ADDITIONAL_INSTANCE_TYPES:
+#  t3.xlarge:      # Required, AWS instance type
+#    price: 0.052  # Required, must be a number
+#    # Remaining fields are optional strings, e.g.
+#    family: 'Burstable'
+#    group: 't3'
+#    vCpu: '4'
+#    mem: '16.00'
+HOTG_ADDITIONAL_INSTANCE_TYPES: []
+
```
Comment (samuelallan72) [inline playbooks/roles/hotg/defaults/main.yml:65]: @pomegranited I just noticed - is it an issue that the default value is a list, while the example and how it is being used it's a dictionary? I think for consistency, the default should be:

```
HOTG_ADDITIONAL_INSTANCE_TYPES: {}
```

Comment (pomegranited) [reply playbooks/roles/hotg/defaults/main.yml:65]: @swalladge That's an excellent catch!  Sorry I missed that.  Fixed with 171eeeb .

**Comment by pomegranited** [comment]
Hi @natabene  -- will this fit into Campus's September review budget?

**Comment by natabene** [comment]
@pomegranited It will budget wise, but not sure about timeline - we only have 5 business days left till it ends.

**Review by fredsmith — APPROVED** [review]

**Comment by fredsmith** [comment]
@pomegranited thanks!

**Comment by itsjeyd** [comment]
@fredsmith Thanks for the review and merge!

**Pull request merged**